build/version.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit -o nounset -o pipefail
+
+git_dirty=$([[ -z $(git status --short) ]] || echo "-dirty")
+git_tag=$(git tag --points-at HEAD)
+if [ -z "${git_tag}" ]; then
+    # No git tag found for current commit, use git describe to construct the
+    # version.
+    git_tag=$(git describe --tags)
+fi
+
+# Use the git tag, removing the leading 'v' if it exists.
+echo "${git_tag/#v/}${git_dirty}"
```

Comment:
Isn't the whole thing just
```bash
tag="$(git describe --tags --dirty)"
echo "${tag#v}"
```

If the current commit is a tag, it just points out the tag (with possibly with a `-dirty` suffix); if it's not, it basically goes through the same flow… See [here](https://repl.it/repls/ExperiencedStupendousCarrier#main.sh) for a test.

Reply:
It doesn't take into account new files not staged.

Reply:
That just affects `${git_dirty}` right? In that case we have
```bash
git_dirty=$([[ -z "$(git status --short)" ]] || echo "-dirty")
describe="$(git describe --tags)"
echo "${describe#v}${git_dirty}"
```
(That is, I don't see why we need the `[ -z "${git_tag}" ]` check.)

Reply:
Yeah ... just a script evolution. :laughing: 
Done with https://github.com/kubernetes-sigs/minibroker/pull/84/commits/c3176cce8e7aa15142c97a61fac3a9b250e3fc52.